tamr_unify_client/dataset/resource.py

```diff
@@ -64,20 +64,21 @@ def attributes(self):
         alias = self.api_path + "/attributes"
         return AttributeCollection(self.client, alias)
 
-    def _update_records(self, updates, **json_args):
+    def _update_records(self, updates, ignore_nan=False):
         """Send a batch of record creations/updates/deletions to this dataset.
         You probably want to use :func:`~tamr_unify_client.dataset.resource.Dataset.upsert_records`
         or :func:`~tamr_unify_client.dataset.resource.Dataset.delete_records` instead.
 
         :param records: Each record should be formatted as specified in the `Public Docs for Dataset updates <https://docs.tamr.com/reference#modify-a-datasets-records>`_.
         :type records: iterable[dict]
-        :param `**json_args`: Arguments to pass to the JSON `dumps` function, as documented `here <https://simplejson.readthedocs.io/en/latest/#simplejson.dumps>`_.
-            Some of these, such as `indent`, may not work with Tamr.
+        :param ignore_nan: Whether to treat `NaN` values as null.  Unconverted `NaN`s will raise an error if found.
```

Comment:
Should we tell users to stop relying on `ignore_nan`? Wondering if we want to deprecate and later remove this parameter.

Alternatively, we could simply say that TUC itself is now deprecated.

Reply:
Something as simple as "WARN: 'ignore_nan' is deprecated. Users are expected to provide valid JSON representations instead" or something like that.

Reply:
Any preference between `DeprecationWarning` and `PendingDeprecationWarning`?

Reply:
`DeprecationWarning` is what we want since there's already viable alternatives.